src/backend/distributed/commands/table.c

```diff
@@ -3490,6 +3490,23 @@ ErrorIfUnsupportedAlterTableStmt(AlterTableStmt *alterTableStatement)
 			{
 				if (IsForeignTable(relationId))
 				{
+					List *optionList = (List *) command->def;
+
+					if (IsCitusTableType(relationId, CITUS_LOCAL_TABLE) &&
+						ForeignTableDropsTableNameOption(optionList))
```

Comment:
shouldn't this only apply to postgres_fdw? Now, we apply the same check for all fdws, which is too restrictive. 

We only strictly require this check for postgres_fdw, see https://github.com/citusdata/citus/pull/6652/files#diff-0e4644d73762dfee1e32c6ac69a46e0802b4675316f466d2c86407464e2cf990R549 and https://github.com/citusdata/citus/pull/6652/files#r1088709237 for some relevant code / discussions